src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java

```diff
@@ -390,6 +453,33 @@ public JSONObject getSlaveAttributeForLabel(String labelName) {
     return null;
   }
 
+  protected Object readResolve() {
```

Comment:
It's really hard to understand what's happening in this method. Even the name of the method doesn't give me a strong clue. This could use some comments.